sw/device/tests/entropy_src_edn_reqs_test.c

```diff
@@ -0,0 +1,223 @@
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rv_core_ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/otbn.h"
+#include "sw/device/lib/testing/aes_testutils.h"
+#include "sw/device/lib/testing/alert_handler_testutils.h"
+#include "sw/device/lib/testing/entropy_testutils.h"
+#include "sw/device/lib/testing/keymgr_testutils.h"
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "alert_handler_regs.h"  // Generated.
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTBN_DECLARE_APP_SYMBOLS(randomness);
+OTBN_DECLARE_SYMBOL_ADDR(randomness, rv);
+OTBN_DECLARE_SYMBOL_ADDR(randomness, fail_idx);
+OTBN_DECLARE_SYMBOL_ADDR(randomness, rnd_out);
+OTBN_DECLARE_SYMBOL_ADDR(randomness, urnd_out);
+
+static const otbn_app_t kOtbnAppCfiTest = OTBN_APP_T_INIT(randomness);
+static const otbn_addr_t kVarRv = OTBN_ADDR_T_INIT(randomness, rv);
+static const otbn_addr_t kVarFailIdx = OTBN_ADDR_T_INIT(randomness, fail_idx);
+static const otbn_addr_t kVarRndOut = OTBN_ADDR_T_INIT(randomness, rnd_out);
+static const otbn_addr_t kVarUrndOut = OTBN_ADDR_T_INIT(randomness, urnd_out);
+
+const test_config_t kTestConfig;
+
+static void aes_test(dif_aes_t *aes) {
+  // Initialise AES.
+  CHECK_DIF_OK(
+      dif_aes_init(mmio_region_from_addr(TOP_EARLGREY_AES_BASE_ADDR), aes));
+
+  LOG_INFO("%s", __func__);
+  // Setup ECB encryption transaction.
+  dif_aes_transaction_t transaction = {
+      .operation = kDifAesOperationEncrypt,
+      .mode = kDifAesModeEcb,
+      .key_len = kDifAesKey128,
+      .manual_operation = kDifAesManualOperationManual,
+      .key_provider = kDifAesKeySideload,
+  };
+
+  // Write the initial key share, IV and data in CSRs (known combinations).
+  CHECK_DIF_OK(dif_aes_start(aes, &transaction, NULL, NULL));
+
+  // Write the PRNG_RESEED bit to reseed the internal state of the PRNG.
+  CHECK_DIF_OK(dif_aes_trigger(aes, kDifAesTriggerPrngReseed));
+}
+
+static void kmac_test(void) {
+  // Intialize KMAC hardware.
+  dif_kmac_t kmac;
+  dif_kmac_operation_state_t kmac_operation_state;
+
+  LOG_INFO("%s", __func__);
+  CHECK_DIF_OK(
+      dif_kmac_init(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR), &kmac));
+
+  // Configure KMAC hardware using software entropy.
+  dif_kmac_config_t config = (dif_kmac_config_t){
+      .entropy_mode = kDifKmacEntropyModeEdn,
+      .entropy_fast_process = kDifToggleDisabled,
+  };
+  CHECK_DIF_OK(dif_kmac_configure(&kmac, config));
+
+  CHECK_DIF_OK(dif_kmac_trigger(&kmac, kDifKmacTriggerEntropyReq));
+
+  dif_kmac_status_t status;
+  //   do {
+  CHECK_DIF_OK(dif_kmac_get_status(&kmac, &status));
+  LOG_INFO("\tStatus=%d", status.sha3_state);
+  //   } while (status.sha3_state != kDifKmacSha3StateIdle);
```

Comment:
Please remove dead code as it will not be tested ;)

Reply:
done